ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -475,24 +481,38 @@ private ReplicaMetadataResponse getReplicaMetadataResponse(List<RemoteReplicaInf
    * @throws StoreException
    */
   private Set<StoreKey> getMissingStoreKeys(ReplicaMetadataResponseInfo replicaMetadataResponseInfo,
-      DataNodeId remoteNode, RemoteReplicaInfo remoteReplicaInfo) throws StoreException {
+      DataNodeId remoteNode, RemoteReplicaInfo remoteReplicaInfo) throws Exception {
     long startTime = SystemTime.getInstance().milliseconds();
     List<MessageInfo> messageInfoList = replicaMetadataResponseInfo.getMessageInfoList();
-    List<StoreKey> storeKeysToCheck = new ArrayList<StoreKey>(messageInfoList.size());
+    List<StoreKey> storeKeysToCheck = new ArrayList<>(messageInfoList.size());
     for (MessageInfo messageInfo : messageInfoList) {
       storeKeysToCheck.add(messageInfo.getStoreKey());
       logger.trace("Remote node: {} Thread name: {} Remote replica: {} Key from remote: {}", remoteNode, threadName,
           remoteReplicaInfo.getReplicaId(), messageInfo.getStoreKey());
     }
 
-    Set<StoreKey> missingStoreKeys = remoteReplicaInfo.getLocalStore().findMissingKeys(storeKeysToCheck);
-    for (StoreKey storeKey : missingStoreKeys) {
-      logger.trace("Remote node: {} Thread name: {} Remote replica: {} Key missing id: {}", remoteNode, threadName,
-          remoteReplicaInfo.getReplicaId(), storeKey);
+    Map<StoreKey, StoreKey> originalToConverted = new HashMap<>();
```

Comment:
isn't this the same as `remoteToLocal` without the `null` values? Why isn't that passed in instead of reconstructing it here?